src/azure-cli-core/azure/cli/core/adal_authentication.py

```diff
@@ -135,3 +131,25 @@ def set_token(self):
     def signed_session(self, session=None):
         logger.debug("MSIAuthenticationWrapper.signed_session invoked by Track 1 SDK")
         super().signed_session(session)
+
+
+def _try_scopes_to_resource(scopes):
+    """Wrap scopes_to_resource to workaround some SDK issues."""
+
+    # Track 2 SDKs generated before https://github.com/Azure/autorest.python/pull/239 don't maintain
+    # credential_scopes and call `get_token` with empty scopes.
+    # As a workaround, return None so that the CLI-managed resource is used.
+    if not scopes:
+        logger.debug("No scope is provided by the SDK, use the CLI-managed resource.")
+        return None
+
+    # Track 2 SDKs generated before https://github.com/Azure/autorest.python/pull/745 extend default
+    # credential_scopes with custom credential_scopes. Instead, credential_scopes should be replaced by
+    # custom credential_scopes. https://github.com/Azure/azure-sdk-for-python/issues/12947
+    # As a workaround, remove the first one if there are multiple scopes provided.
+    if len(scopes) > 1:
+        logger.debug("Multiple scopes are provided by the SDK, discarding the first one: %s", scopes[0])
+        return scopes_to_resource(scopes[1:])
```

Comment:
Does the custom credential_scopes always start with index 1 ? And it seems that you return an array when `len(scopes) > 1` while other cases will return a string. The signature in [AuthenticationContext.acquire_token](https://github.com/AzureAD/azure-activedirectory-library-for-python/blob/01f6fad2928c02134def85a2ff1fdaf083e91701/adal/authentication_context.py#L133) says the resource is a string type. So are you sure it can work properly when pass an array to it ?

Reply:
`len(scopes) > 1` is used as a guard to detect the mentioned bug. Ideally, one and only one scope should appear in the tuple.